tee-worker/litentry/core/data-providers/src/graphql.rs

```diff
@@ -179,12 +216,56 @@ impl GraphQLClient {
 			Err(Error::GraphQLError("Invalid GraphQL response".to_string()))
 		}
 	}
+
+	pub fn query_total_transactions(
+		&mut self,
+		credentials: VerifiedCredentialsTotalTxs,
+	) -> Result<Vec<TotalTxsStruct>, Error> {
+		let path = "latest/graphql".to_string();
+		let query_value = credentials.to_graphql();
+		let query = vec![("query", query_value.as_str())];
+		let response = self
+			.client
+			.get_with::<String, QLResponse>(path, query.as_slice())
+			.map_err(|e| Error::RequestError(format!("{:?}", e)))?;
+
+		let mut result: HashMap<String, TotalTxsStruct> = HashMap::new();
+
+		response.data.get("data").and_then(|v| v.as_object()).and_then(|map| {
+			for (_network, list) in map {
+				list.as_array().and_then(|element| {
+					for x in element {
+						// aggregate total_transactions from different networks, like group_by.
+						if let Ok(obj) = serde_json::from_value::<TotalTxsStruct>(x.clone()) {
+							if result.contains_key(&obj.address) {
+								result.get_mut(&obj.address).and_then(|origin| {
+									origin.total_transactions += obj.total_transactions;
+									None::<u8>
+								});
+							} else {
+								result.insert(obj.address.clone(), obj.clone());
+							}
+						}
+					}
+					None::<u8>
+				});
+			}
+			None::<u8>
+		});
+		if !result.is_empty() {
+			// Ok(result.iter().map(|(_, v)| v.clone()).collect::<Vec<TotalTxsStruct>>())
+			Ok(result.values().cloned().collect::<Vec<TotalTxsStruct>>())
+		} else {
+			Err(Error::GraphQLError("Invalid GraphQL response".to_string()))
+		}
```

Comment:
may be "Empty GraphQL response" would be better.

Reply:
Maybe http request get an error(status != 200) or parses the json error, but I ignore these errors in the method, so I think maybe the empty_graphql_response is not accurate. :sweat_smile:

Reply:
so it will not be an error when the client `get_with` response status != 200?

```
self
			.client
			.get_with::<String, QLResponse>(path, query.as_slice())
```

Reply:
It should only have parsing json errors